### DISCUSSION TITLE
Page tree navigation doesn't comply with regexp permission

### DISCUSSION THREAD

**Discussion by enryIT** [requarks/wiki#3670] [Error / Bug Report]
**Describe the bug**
Page tree navigation doesn't comply with regexp group permission

**To Reproduce**
1. Create a group named `testgroup`
2. Give the group `testgroup` a single regexp permission (eg. read pages that match the posix expression `.*public.*`)
3. Create a user named `testuser`
3. Put the `testuser` user in the `testgroup` group
4. Create the page `/test/public/test`
3. The user `testuser` is capable of reading the page using direct link (eg. http://wiki.example.com/test/public/test) but is not seeing the page in the page navigation tree

**Expected behavior**
The `testuser` user is able to see both the page under the navigation menu and using direct link.

**Host Info:**
 - OS: Docker
 - Wiki.js version: 2.3.77
 - Database engine: postgres 12

**Comment by Chrisc-c-c** [comment]
This same issue exists for tag-based permissions.

**Comment by enryIT** [comment]
As a simple workaround you can use the regexp or tag permission only for negation.

For example, supposte you have this structure:
```
root
-- test1
---- private
---- public
-- test2
---- private
---- public
```
and you want to give a specific group access to both private and public for `test1` folder but only public access to `test2`.
You have to grant read permission with "starting with" on `test1/public` and `test1/private` **and** negate read permission with "regexp" `.*private.*`

This workaround only works with simple situations, if you require more complex permission structure this isn't feasible.

@NGPixel have you got any chance to check this issue? This is a real problem with shared wikis.

**Comment by yusufgungor** [comment]
+1
